api/base/exceptions.py

```diff
@@ -80,20 +80,82 @@ class Conflict(APIException):
     default_detail = ('Resource identifier does not match server endpoint.')
 
 
-class InvalidQueryStringError(JSONAPIException):
+class JSONAPIParameterException(JSONAPIException):
+    def __init__(self, detail=None, parameter=None):
+        source = {
+            'parameter': parameter
+        }
+        super(JSONAPIParameterException, self).__init__(detail=detail, source=source)
+
+
+class JSONAPIAttributeException(JSONAPIException):
+    def __init__(self, detail=None, attribute=None):
+        source = {
+            'pointer': '/data/attributes/{}'.format(attribute)
+        }
+        super(JSONAPIAttributeException, self).__init__(detail=detail, source=source)
+
+
+class InvalidQueryStringError(JSONAPIParameterException):
     """Raised when client passes an invalid value to a query string parameter."""
     default_detail = 'Query string contains an invalid value.'
     status_code = http.BAD_REQUEST
 
-    def __init__(self, detail=None, parameter=None):
-        super(InvalidQueryStringError, self).__init__(detail=detail, source={
-            'parameter': parameter
-        })
 
+class InvalidFilterOperator(JSONAPIParameterException):
+    """Raised when client passes an invalid operator to a query param filter."""
+    status_code = http.BAD_REQUEST
+
+    def __init__(self, detail=None, value=None):
+        if value and not detail:
+            detail = "Value '{0}' is not a supported filter operator; use one of eq, lt, lte, gt, gte.".format(value)
+        super(InvalidFilterOperator, self).__init__(detail=detail, parameter='filter')
+
+
+class InvalidFilterValue(JSONAPIParameterException):
+    """Raised when client passes an invalid value to a query param filter."""
+    default_detail = 'Invalid filter value'
+    status_code = http.BAD_REQUEST
+
+    def __init__(self, detail=None, value=None, field_type=None):
+        if value and not detail:
+            detail = "Value '{0}' is not valid for a filter on type {1}.".format(
```

Comment:
Also, this will display `...for a filter on type None` if `field_type` isn't passed; either make `field_type` a required argument or modify the error message if `field_type` isn't passed.